src/ClimaEarth.jl

```diff
@@ -1,5 +1,233 @@
+"""
+function interpolate_atmosphere_state!(interfaces, 
+                                       atmosphere::CA.AtmosSimulation, 
+                                       coupled_model)
+
+    remapper = interfaces.exchanger.atmosphere_exchanger
+    exchange_atmosphere_state = interfaces.exchanger.exchange_atmosphere_state
+
+    ue  = parent(exchange_atmosphere_state.u)
+    ve  = parent(exchange_atmosphere_state.v)
+    Te  = parent(exchange_atmosphere_state.T)
+    qe  = parent(exchange_atmosphere_state.q)
+    pe  = parent(exchange_atmosphere_state.p)
+
+    ue = dropdims(ue, dims=3)
+    ve = dropdims(ve, dims=3)
+    Te = dropdims(Te, dims=3)
+    qe = dropdims(qe, dims=3)
+    pe = dropdims(pe, dims=3)
+
+    Uah = CC.Geometry.UVVector.(CC.Spaces.level(atmosphere.integrator.u.c.uₕ, 1))
+    ua = Uah.components.data.:1
+    va = Uah.components.data.:2
+
+    tsa = CC.Spaces.level(atmosphere.integrator.p.precomputed.ᶜts, 1)
+    ℂa = atmosphere.integrator.p.params.thermodynamics_params
+    Ta = Thermodynamics.air_temperature.(ℂa, tsa)
+    pa = Thermodynamics.air_pressure.(ℂa, tsa)
+    qa = Thermodynamics.total_specific_humidity.(ℂa, tsa)
+
+    #=
+    # TODO: make this work without allocation
+    #       make sure that Remapper(args...; buffer_length=5)
+    #       or whatever it needs to be
+    exchange_fields = cat(ue, ve, Te, pe, qe, dims=3)
+    atmos_fields    = [ua, va, Ta, pa, qa]
+    CC.Remapping.interpolate!(exchange_fields, remapper, atmos_fields)
+    =#
+
+    CC.Remapping.interpolate!(ue, remapper, ua)
+    CC.Remapping.interpolate!(ve, remapper, va)
+    CC.Remapping.interpolate!(Te, remapper, Ta)
+    CC.Remapping.interpolate!(pe, remapper, pa)
+    CC.Remapping.interpolate!(qe, remapper, qa)
+
+    #=
+    # This is needed, unless the above computations include the halos.
+    # OC.fill_halo_regions!(exchange_atmosphere_state.u)
+    # OC.fill_halo_regions!(exchange_atmosphere_state.v)
+    # OC.fill_halo_regions!(exchange_atmosphere_state.T)
+    # OC.fill_halo_regions!(exchange_atmosphere_state.q)
+    # OC.fill_halo_regions!(exchange_atmosphere_state.p)
+    =#
+
+    return nothing
+end
+
+# Note: this just copies, for now.
+@kernel function _interpolate_atmosphere_state!(exchange_state, atmos_state)
+    i, j = @index(Global, NTuple)
+    @inbounds begin
+        exchange_state.u[i, j, 1] = atmos_state.u[i, j]
+        exchange_state.v[i, j, 1] = atmos_state.v[i, j]
+        exchange_state.T[i, j, 1] = atmos_state.T[i, j]
+        exchange_state.q[i, j, 1] = atmos_state.q[i, j]
+        exchange_state.p[i, j, 1] = atmos_state.p[i, j]
+    end
+end
+
+function atmosphere_exchanger(atmosphere::CA.AtmosSimulation, exchange_grid)
+    λ = λnodes(exchange_grid, Center(), Center(), Center(), with_halos=true)
+    φ = φnodes(exchange_grid, Center(), Center(), Center(), with_halos=true)
+
+    if exchange_grid isa LatitudeLongitudeGrid
+        λ = reshape(λ, length(λ), 1)
+        φ = reshape(φ, 1, length(φ))
+    end
```

Comment:
hm..

````julia
nodes(exchange_grid, Center(), Center(), Center(), with_halos=true, reshape=true)
```

might be easier to generalize here to any grid?

Reply:
Finally just to note that we can't do Tripolar yet because we don't have a way to interpolate from ClimaAtmos to tripolar. Hopefully eventually we'll have that.

Reply:
Gotcha!